CodeU_Assignement_5/src/codeu_assignement_5/KemUnknownLanguage.java

```diff
@@ -0,0 +1,145 @@
+package codeu_assignement_5;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author Karine
+ */
+public class KemUnknownLanguage {
+    /*
+     * Input: a dictionary (a list of words in lexicographic order) of all words in an unknown/invented language, 
+    *  Output: alphabet (an ordered list of characters) of that language, letters we cannot decide are appended in the end.
+    */
+    public List<Character> getAlphabeit(String[] alienWords) {
+        if ((alienWords != null) && (alienWords.length > 0)) {
+           return extractAlphabeitFromWords(alienWords);
+        }
+        
+        return null;
+    }
+
+    private List<Character> extractAlphabeitFromWords(String[] alienWords) {
+        /* Get the current order by prefix size, starting from 0 */
+        List<Character> dict = getBasicAlphaBeit(alienWords, 0); /* Basic Alphabeit */ 
+        /* A,R,C from the example! */
+
+        /* Add from prefix while there are letters after first letter of prefix */
+        while (!addToDictByPrefix(alienWords, dict)){} 
+
+        /* Add all the chars we cannot decide in the end*/
+        for (String w : alienWords) {
+            for (Character c: w.toCharArray()) {
+                if (!dict.contains(c)) {
+                    dict.add(c);
+                }
+            }
+        }
+        
+        return dict;
+    }
+    
+    
+    /*
+     * Inputs: All words with the same prefix (all the first prefixSize chars are the same!)
+     * Output: Per common prefix, gets the order of the chars
+     *
+     * Note: Assume the input is valid here!
+    */
+    private List<Character> getBasicAlphaBeit(String[] words, int prefixSize) {
```

Comment:
This is one of the clearer methods, but it's still a bit messy.

Perhaps call it "inferAlphabetFromCommonPrefix"?
`
/* Given an array of words that all match up to a common prefix length, infer an alphabetical ordering of the characters based on the order of the first differing character. No check is made that the common prefix really is the same across the list. */
`